golem/core/keysauth.py

```
@@ -182,6 +185,11 @@ def _generate_keys(difficulty: int) -> Tuple[bytes, bytes]:
             if KeysAuth.is_pubkey_difficult(pub_key, difficulty):
                 break
 
+            # lets be responsive to reactor stop (eg. ^C hit by user)
+            if reactor_started and not reactor.running:
+                logger.warning("reactor stopped, aborting key generation ..")

```

No need for log message if you're throwing an exception - it's gonna be caught somewhere and logged anyway

---

nope. it is not caught because reactor is stopped...

---

nope

---

So it's silently swallowed? That doesn't sound right

---

please notice that this exception is thrown **_after_** reactor is stopped, so no callback nor errback would be executed. I'm throwing it to break processing. It could be `return`, but i think `raise` is more natural in this situation. Logger is additional, to be informative to our users.